pkg/opscomments/comments.go

```diff
@@ -0,0 +1,161 @@
+package opscomments
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/info"
+)
+
+var (
+	testAllRegex      = regexp.MustCompile(`(?m)^/test\s*$`)
+	retestAllRegex    = regexp.MustCompile(`(?m)^/retest\s*$`)
+	testSingleRegex   = regexp.MustCompile(`(?m)^/test[ \t]+\S+`)
+	retestSingleRegex = regexp.MustCompile(`(?m)^/retest[ \t]+\S+`)
+	oktotestRegex     = regexp.MustCompile(`(?m)^/ok-to-test\s*$`)
+	cancelAllRegex    = regexp.MustCompile(`(?m)^(/cancel)\s*$`)
+	cancelSingleRegex = regexp.MustCompile(`(?m)^(/cancel)[ \t]+\S+`)
+)
+
+type EventType string
+
+func (e EventType) String() string {
+	return string(e)
+}
+
+var (
+	NoOpsCommentEventType        = EventType("no-ops-comment")
+	TestCommentEventType         = EventType("test-comment")
+	RetestSingleCommentEventType = EventType("retest-comment")
+	RetestAllCommentEventType    = EventType("retest-all-comment")
```

Comment:
I see SingleCommentEventType and  AllCommentEventType for `retest` & `cancel` but not for `test`
any reason ?
asking because `test` also works similar to `retest`

Reply:
issuing a single /test doesn't issue a /retest of all pipelines, only /retest does afaik,

Reply:
ah yeah it does, i think the initial design we said that only retest retest all (like prow) and the previous code says it as much by not defining a TestAll Regexp, i guess it's a side effect we have to live forever....

Reply:
btw what is the difference between `retest` and `test` ? because both comment will trigger the CI 
So i am not able to see difference between those 2  :thinking:

Reply:
it's a categorization to know where things come from if the pr got started with /retest it would be set in the eventtype or /test

I was thinking to remove it and  remove the retest type all together but i think it's fine to keep it since that makes things less surprising